src/main/java/io/cos/cas/osf/model/OsfUser.java

```diff
@@ -86,6 +90,10 @@ public boolean isConfirmed() {
         return dateConfirmed != null;
     }
 
+    public boolean isTermsOfServiceAccepted() {
+        return dateTermsOfServiceAccepted != null;
```

Comment:
Just to verify my own understanding: since this is the first version with this feature it's sufficient to test that it is `NOT NULL`.  But for future updates of the TOS, this code will need to be updated to test against a provided timestamp.  How are we planning to provide that timestamp?  Will it be hard-coded, loaded from an external resource, or something else?  Not a blocker, just curious.

Reply:
Good question!  As far as I know, OSF only has this one field for TOS. The only way for OSF to ask user to agree again after TOS updated is to set this field NULL ...